opm/simulators/flow/BlackoilModelNldd.hpp

```diff
@@ -815,7 +815,7 @@ class BlackoilModelNldd {
                            const SimulatorTimerInterface& timer,
                            const Domain& domain)
     {
-        auto initial_local_well_primary_vars = model_.wellModel().getPrimaryVarsDomain(domain);
+        auto initial_local_well_primary_vars = model_.wellModel().getPrimaryVarsDomain(domain.index);
```

Comment:
Why did you change the interface from taking a Domain to taking an index?

Reply:
because Domain is dependent on the full typetag.